site/_docs/reference.md

```diff
@@ -1871,14 +1872,44 @@ is named as "fixed windowing".
 
 | Operator syntax      | Description
 |:-------------------- |:-----------
-| TUMBLE(table, DESCRIPTOR(column_name), interval [, time ]) | Indicates a tumbling window of *interval* for *datetime*, optionally aligned at *time*. Tumbling is applied on table in which there is a watermarked column specified by descriptor.
+| TUMBLE(table, DESCRIPTOR(datetime), interval [, time ]) | Indicates a tumbling window of *interval* for *datetime*, optionally aligned at *time*.
 
 Here is an example:
 `SELECT * FROM TABLE(TUMBLE(TABLE orders, DESCRIPTOR(rowtime), INTERVAL '1' MINUTE))`,
 will apply tumbling with 1 minute window size on rows from table orders. rowtime is the
 watermarked column of table orders that tells data completeness.
 
+#### HOP
+In streaming queries, HOP assigns windows that cover rows within the interval of *size*, shifting every *slide*, 
+and optionally aligned at *time* based on a timestamp column. Windows assigned could have overlapping so hopping 
+sometime is named as "sliding windowing".  
+
+
+| Operator syntax      | Description
+|:-------------------- |:-----------
+| HOP(table, DESCRIPTOR(datetime), slide, size, [, time ]) | Indicates a hopping window for *datetime*, covering rows within the interval of *size*, shifting every *slide*, and optionally aligned at *time*.
+
+Here is an example:
+`SELECT * FROM TABLE(HOP(TABLE orders, DESCRIPTOR(rowtime), INTERVAL '2' MINUTE, INTERVAL '5' MINUTE))`,
+will apply hopping with 5-minute interval size on rows from table orders, shifting every 2 minutes. rowtime is the
+watermarked column of table orders that tells data completeness.
+
+#### SESSION
+In streaming queries, SESSION assigns windows that cover rows based on *datetime*. Within a session window, distances 
+of rows are less than *interval*. Session window is applied per *key*.
+
+
+| Operator syntax      | Description
+|:-------------------- |:-----------
+| session(table, DESCRIPTOR(datetime), DESCRIPTOR(key), interval [, time ]) | Indicates a session window of *interval* for *datetime*, optionally aligned at *time*. Session window is applied per *key*.
+
+Here is an example:
+`SELECT * FROM TABLE(SESSION(TABLE orders, DESCRIPTOR(rowtime), DESCRIPTOR(product), INTERVAL '20' MINUTE))`,
+will apply session with 20-minute inactive gap on rows from table orders. rowtime is the
+watermarked column of table orders that tells data completeness. Session is applied per product.
+
 ### Grouped window functions
```

Comment:
Some comments:
1) Is it necessary to deprecate the manner of using them as grouped window function?
2) Update "Table-valued functions." to "Table functions".  
3) Could you add doc the [streaming](https://calcite.apache.org/docs/stream.html) page? I think we can do this work in this PR or after finishing the umbrella JIRA.

Reply:
Re: 1. Yes. First of all, Table function style streaming can cover what grouped window function provides. Then grouped window function has drawbacks: 1) magic *_start, * _end function to get window_start, window_end 2) enforce to us a GROUP BY (if users only want to do windowing streaming join without aggregation, they shouldn't be bothered to use a GROUP BY). 3) error prone syntax (why users cannot do GROUP BY TUMBLE(), HOP()? The syntax is correct for SQL).
So I think (and probably Julian also agreed) grouped window function can be deprecated. 

I will send an email to discuss the deprecation with community. So grouped window functions won't disappear silently. 


Re 2 and 3: nice suggestions. I will add a commit for the change.

Reply:
I have addressed 2.

Regarding 3, I would do it later. I would prefer to have a discussion about deprecating grouped window functions in mailing list first, which will give a much better idea how `streaming` page be updated (e.g. keep both style or just keep the new one)